dask/dataframe/tests/test_shuffle.py

```diff
@@ -602,6 +602,9 @@ def test_set_index_sorted_true():
     with pytest.raises(ValueError):
         a.set_index(a.z, sorted=True)
 
+    with pytest.raises(ValueError):
+        a.set_index(a.y, sorted=True)
```

Comment:
Ideally this would still work correctly, and we would move the extra data from one partition over to the neighboring one with a little bit of communication.

If we accept the change that you've proposed then we probably don't want to enforce this as a test, but might instead prefer an xfailed test that ensures that the resulting divisions are valid, even if the graph we have to produce is a little bit more complex.

Reply:
It may also be that we want to accept our current incorrect behavior rather than err.  My guess is that in most cases where this happens today it doesn't negatively affect users.  This is hard to judge though/

Reply:
Seems pretty complicated. :)

Perhaps I could just change this to a warning.

Reply:
It could also be that you could leave the existing semi-invalid divisions, and then call `repartition(divisions=...)` with a suitable new suggestion.   I suspect that that would work today and may not be difficult to implement.

Reply:
I tried that and it does not appear to work (without modifying `repartition_divisions` or creating a new method of repartitioning specific to this almost-partitioned case).